vercel.json

```
@@ -1,33 +1,21 @@
 {
-  "buildCommand": "npm run build  > >(tee -a build-output.log) 2> >(tee -a build-output.log >&2) && ! (cat build-output.log | grep -qi error) && echo \"No errors found in build output\"",
+  "buildCommand": "scripts/build-vercel-output.sh",
+  "installCommand": "npm install --no-fund --no-audit",
+  "redirects": [
+    { "source": "/package.json", "destination": "/404" },
+    { "source": "/prerender/:path*", "destination": "/404" },
+    { "source": "/", "destination": "/catalog" },
+    { "source": "/join_team", "destination": "https://backend.codecrafters.io/join_team" },
+    { "source": "/setup_team", "destination": "https://backend.codecrafters.io/setup_team" },
+    { "source": "/r/famous-wombat-417894", "destination": "/join?via=gnomezgrave" }
+  ],
   "rewrites": [
     { "source": "/submissions", "destination": "https://backend.codecrafters.io/submissions" },
     { "source": "/submissions/:match*", "destination": "https://backend.codecrafters.io/submissions/:match*" },
     { "source": "/progress/:match*", "destination": "https://backend.codecrafters.io/progress/:match*" },
+    { "source": "/users/:suffix", "destination": "/prerender/users/user" },
```

The destination path convention is not super clear to me here, can we have it be based on the ember route name directly? i.e. `/prerender/user`.

---

I wonder if a way to be more explicit here would be to do:

```suggestion
    { "source": "/users/:suffix", "destination": "/prerender/user?prefix=/users" },
```

That way it's more clear what "prefix" and "suffix" mean, and the naming convention is more clearly based on ember route names

---

So the trick is, I want to function to be able to handle both types of routes:

```
/users/:suffix --> resolves to /users/user.func — wildcard function that gets passed `suffix` as a queryParam
/demo/code-mirror --> resolves to /demo/code-mirror.func — no wildcard, no suffix, just render a direct route
```

I can add better names for suffix & prefix, for example `_prerenderModelId` and `prerenderPrefix`, and add JSDoc to the `extractPath` function.

This approach is better than passing `prefix=...` manually, because it works automatically and you don't have to pass the prefix.

---

Oh, okay - yeah got it. So the convention is just based on paths then (instead of ember routes), makes sense.

Let's go ahead and merge this, will think about an alternative that is slightly less confusing - still think the prefix/suffix stuff is pretty indirect and could be cleaned up. Nothing comes to mind right now but I'm sure there's a way out here...

---

Btw, I chose `prefix` and `suffix` because in theory there should also be `ending`: 

If I want to rewrite `/demo/code-mirror?param1&param2&param3&...`, to my prerender function and pass those query params to `visit()` — then `ending` wound be the query param part: `?param1&...`

---

Found a better approach, using function names wrapped in `[]`, for example `/prerender/users/[user].func`, to denote "wildcard" functions, as done [in this vercel example](https://github.com/vercel/examples/tree/main/build-output-api/on-demand-isr/.vercel/output/functions/blog). Updated `parsePrerenderPath` to reflect this.